src/main/java/com/internet/shop/dao/impl/UserDaoImpl.java

```diff
@@ -40,4 +40,11 @@ public User update(User user) {
     public boolean deleteById(Long id) {
         return Storage.users.removeIf(user -> user.getId().equals(id));
     }
+
+    @Override
+    public Optional<User> findByLogin(String login) {
+        return Storage.users.stream()
+                .filter(s -> s.getLogin().equals(login))
```

Comment:
`s` is not informative

Reply:
Done. Renamed s. Now this variable is called user.